### PR TITLE
Car Rental: Unit integration `Car` and fix n+1 query

### PR DESCRIPTION
**Pull request by wassimchakib**
## Summary

- In this feature, I added unit tests and integration test for Car model

## Changes log

- [x] Fixed Car Controller
- [x] Added `seeds.rb`
- [x] Fixed n+1 query
- [x] Added unit test for `Car`
- [x] Added Integration test for `CarsController`

## Acceptance criteria

- [x] The tests should cover all edge cases
- [x] The tests should have enough cases to test

### CODE REVIEW

File: app/controllers/api/v1/cars_controller.rb [hunk 1]
```diff
@@ -1,43 +1,82 @@
 class Api::V1::CarsController < ApplicationController
+  # api/v1/cars
   def index
-    @cars = Car.all
-    render json: @cars
+    list_cars = Car.includes(:images).all.as_json(include: :images)
```
Comment (Kweeka1) [inline app/controllers/api/v1/cars_controller.rb:4]: - calling `.all` has no value here.

File: app/controllers/api/v1/cars_controller.rb [hunk 2]
```diff
@@ -1,43 +1,82 @@
...
 
+  # api/v1/cars/{id}
   def show
-    @car = Car.find(params[:id])
-    render json: @car
+    car = Car.find_by(id: params[:id])
+    if car
+      render json: {
+        data: {
+          cars: Car.find_by(id: params[:id]).as_json(include: :images)
```
Comment (Kweeka1) [inline app/controllers/api/v1/cars_controller.rb:18]: - Although rails will cache the first query for a second use. It would be better if you run a single query to get the car record.
```ruby
    car = Car.find_by(id: params[:id])
    if car
      render json: {
        data: {
          cars: car.as_json(include: :images)
        }
      }, status: :ok
```

File: app/controllers/api/v1/reservations_controller.rb [hunk 3]
```diff
@@ -1,14 +1,21 @@
 class Api::V1::ReservationsController < ApplicationController
   def index
+    list_reservations = Reservation.includes(car: [:images]).all
```
Comment (Kweeka1) [inline app/controllers/api/v1/reservations_controller.rb:3]: - Calling `.all` has no value here.

File: app/controllers/api/v1/reservations_controller.rb [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
+                                               } })
+
     render json: {
       data: {
-        reservations: Reservation.all
-      }
-    }, status: :ok
+        reservations: list_reservations
+      }, status: :ok
+    }
```
Comment (Kweeka1) [inline app/controllers/api/v1/reservations_controller.rb:14]: - I think `status: :ok` is meant to apply 200 status code to response. However it is rendered as json property along with reservation list.

**Review by Kweeka1 — APPROVED** [review]
Hi @wassimchakib, 

Your task is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
:heavy_check_mark: Good job refactoring code.
:heavy_check_mark: Tests are running.
:heavy_check_mark: Good PR summary.

## Optional suggestions
- Please check optional suggestions below.

Cheers and Happy coding!👏👏👏
